frontend/src/components/Map.tsx

```diff
@@ -1,82 +1,101 @@
-import MapView, { type Region, PROVIDER_GOOGLE, type Details } from 'react-native-maps'
-import { StyleSheet, type ViewProps, StatusBar } from 'react-native'
+import MapView, { PROVIDER_GOOGLE } from 'react-native-maps'
+import { View, StyleSheet, type ViewProps, StatusBar, type StyleProp, type ViewStyle } from 'react-native'
 import { type Coordinate } from '../services/location'
-import { useRef, useState, useMemo } from 'react'
-
-const GOLDEN: Region = {
-  latitude: 39.749675,
-  longitude: -105.222606,
-  latitudeDelta: 0.005,
-  longitudeDelta: 0.005
-}
+import { LocationButton } from './LocationButton'
+import React, { useRef, useMemo, useState } from 'react'
 
 /**
- * Wraps the expo {@interface MapView} with additional functionality.
+ * A wrapper around react native {@class MapView} that provides a simplified interface for the purposes of this app.
  */
-export function Map(props: ViewProps & MapProps): React.ReactElement<ViewProps> {
-  const [userRegionChanged, setUserRegionChanged] = useState(false)
+export function Map(props: MapProps & ViewProps): React.ReactElement<MapProps & ViewProps> {
   const mapRef = useRef<MapView>(null)
+  const [followingLocation, setFollowingLocation] = useState<boolean>(true)
+  const [lastLocation, setLastLocation] = React.useState<Coordinate | undefined>(undefined)
 
   function panToLocation(location: Coordinate | undefined): void {
-    // We want to make sure we won't snap back to the user location if they decide to pan around,
-    // so check if that's the case before panning.
-    if (location !== undefined && mapRef.current != null && !userRegionChanged) {
+    if (location !== undefined && mapRef.current != null) {
       mapRef.current.animateCamera({
         center: location,
         zoom: 17
       })
     }
   }
 
-  function handleRegionChange(details: Details): void {
-    // If the user is panning around, we don't want to snap back to their location. Note that we
-    // make sure to exclude camera pans from this to avoid disabling location following at soon
-    // as it changes.
-    // details.isGesture is only unavailable on apple maps (which we aren't using)
-    // We default to true because we don't want to lock the map if something is broken
-    if (details.isGesture ?? true) {
-      setUserRegionChanged(true)
+  function updateLocation(location: Coordinate | undefined): void {
+    // Required if we need to re-toggle location following later, which requires
+    // us to pan the camera to whatever location we were last given. Always track
+    // this regardless of if following is currently on.
+    setLastLocation(location)
+    if (followingLocation) {
+      panToLocation(location)
     }
   }
 
+  function flipFollowingLocation(): void {
+    const newState = !followingLocation
+    setFollowingLocation(newState)
```

Comment:
need to use an arrow function in setFollowingLocation to get old value instead of getting from the state (otherwise we might have annoying react state issues)

setFollowingLocation((old) => (!old)) for example